src/plugins/notebook/components/NotebookEntry.vue

```diff
@@ -205,13 +216,45 @@ export default {
             }
         }
     },
+    data() {
+        return {
+            editMode: false,
+            canEdit: true
+        };
+    },
     computed: {
         createdOnDate() {
             return this.formatTime(this.entry.createdOn, 'YYYY-MM-DD');
         },
         createdOnTime() {
             return this.formatTime(this.entry.createdOn, 'HH:mm:ss');
         },
+        formattedText() {
+            let text = this.entry.text;
+
+            if (this.editMode) {
+                return sanitizeHtml(text);
+            }
+
+            let urlsExist = false;
+            let urlMap = {};
+
+            while (text.match(URL_REGEX)) {
```

Comment:
I _think_ we can do this with a single greedy `.replace` and a callback.

eg.
``` JavaScript
text = text.replace(URL_REGEX, (match) => {
    //do the replacement
});
```

You could do it without a function just using a group substitution, but I'd like to restrict to allowed domains as well (see comment on this), and that logic will be easier to put in a function than compiling a custom mega-regex.

Reply:
haha, this is SO much simpler. Thank you!